cep-9999.md

```diff
@@ -0,0 +1,94 @@
+# CEP XXXX - Names in conda packages and channels
+
+<table>
+<tr><td> Title </td><td> CEP XXXX - Names in conda packages and channels </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt; <br /> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Mar 11, 2025</td></tr>
+<tr><td> Updated </td><td> Mar 11, 2025</td></tr>
+<tr><td> Discussion </td><td> N/A </td></tr>
+<tr><td> Implementation </td><td> N/A </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to standardize names and other strings used throughout the conda ecosystem.
+
+## Specification
+
+### Artifacts
+
+The conda ecosystem distinguishes between two types of packages:
+
+- Distributable package names: represented by a concrete, downloadable, extractable conda artifact.
+- Virtual package names: not backed by any physical artifact. They only exist on the client side.
+
+### Package names
+
+A distributable package name MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. It MUST start with a letter, a number, or a single underscore. It MUST NOT include two consecutive separators (hyphen, period, underscore).
+
+Virtual package names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. They MUST NOT use two consecutive separators, with one exception: they MUST start with two underscores.
+
+This means that distributable package names MUST match the following case-insensitive regex: `^(([a-z0-9])|([a-z0-9_](?!_)))[._-]?([a-z0-9]+(\.|-|_|$))*$`.
```

Comment:
As of 2025-03-12T19:00Z, of the ~209K unique distributeable [conda] package names on anaconda.org, only 3 violate the `^[a-z0-9_][a-z0-9_.-]*$` naming pattern:
* one is the `__anaconda_core_depends` package Anaconda is working on renaming;
* one is the name of an existing virtual package under a "hack" label; and
* one contains a space but is a broken artifact with zero downloads

(I'm still looking for package names with consecutive separators; the database I'm querying doesn't support backreferences in its regexps.)